fix usage string for the kubectl logs command

tvieira opened the pull request:
Even though the use of an inline [CONTAINER] name is still accepted for,
legacy purpose the error message does not match what the documentation
or man page says. This commit aligns the usage string that is displayed
when the `kubectl logs` command is called with more than one container
name (with the use of the -c flag or not).

```release-note
NONE
```

Commit:
fix usage string for the kubectl logs command

Even though the use of an inline [CONTAINER] name is still accepted,
this does not match what the documentation or man page says. This commit
aligns the usage string that is displayed when the kubectl logs command
is called with more than one container name (with the use of flag or
not).

nikhita commented on pkg/kubectl/cmd/logs.go:
```diff
@@ -71,7 +71,7 @@ var (
 
 const (
-	logsUsageStr = "expected 'logs (POD | TYPE/NAME) [CONTAINER_NAME]'.\nPOD or TYPE/NAME is a required argument for the logs command"
+	logsUsageStr = "expected 'logs [-f] [-p] (POD | TYPE/NAME) [-c CONTAINER]'.\nPOD or TYPE/NAME is a required argument for the logs command"
```
:+1:

ceshihao replied:
Perhaps, you can reuse the string here.
https://github.com/kubernetes/kubernetes/blob/c4ca6638c397e90a486d363a10b3e6411fa6a0e4/pkg/kubectl/cmd/logs.go#L103

You do not have to modify multiple strings if the usage is changed again.

nikhita commented:
/ok-to-test

nikhita commented:
@tvieira thanks for your first PR to Kubernetes! :)

tvieira commented:
/retest

ceshihao commented:
/sig cli

soltysh commented on pkg/kubectl/cmd/logs_test.go:
```diff
@@ -196,7 +196,7 @@ func TestLogComplete(t *testing.T) {
 			name:     "No args case",
 			flags:    map[string]string{"selector": ""},
-			expected: "'logs (POD | TYPE/NAME) [CONTAINER_NAME]'.\nPOD or TYPE/NAME is a required argument for the logs command",
+			expected: "'logs [-f] [-p] (POD | TYPE/NAME) [-c CONTAINER]'.\nPOD or TYPE/NAME is a required argument for the logs command",
```
Honestly, these tests are worthless, they don't give any value. We don't guarantee output stability and this is testing just that. Please remove both.

tvieira commented:
Sorry the delay folks, a mixed of conferences & time off delayed this commit.

@ceshihao I created then a usage string that contains the usage format so it can be used in other places (such as doc), and replaced the previous string with a usage error string, which reuses the newly created usage format to display how it should be used.

@soltysh I removed both tests.

tvieira commented:
/test pull-kubernetes-integration

tvieira commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu

tvieira commented:
Hi @ceshihao , would you mind to review the changes based on your suggestion?

tvieira commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu

tvieira commented:
Hi @soltysh, would you mind to review the changes?

dixudx commented on pkg/kubectl/cmd/logs_test.go:
```diff
@@ -245,13 +245,6 @@ func TestLogComplete(t *testing.T) {
 		opts     func(genericclioptions.IOStreams) *LogsOptions
 		expected string
 	}{
-		{
```
Why removing these tests?

tvieira replied:
@dixudx see @soltysh comments above: https://github.com/kubernetes/kubernetes/pull/65545#discussion_r199498125

soltysh commented on pkg/kubectl/cmd/logs.go:
```diff
@@ -195,7 +196,7 @@ func (o *LogsOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []str
 	case 0:
 		if len(o.Selector) == 0 {
-			return cmdutil.UsageErrorf(cmd, "%s", logsUsageStr)
+			return cmdutil.UsageErrorf(cmd, "%s", logsUsageErrStr)
```
Don't, just print the error you think is valuable here, eg. `selector is required`, which btw. can be done with `cmd.MarkFlagRequired` function.

soltysh commented:
/approve

dixudx reviewed (approved):
/lgtm
